crypto_kem/ledacrypt23371/clean/niederreiter_keygen.c

```diff
@@ -0,0 +1,76 @@
+#include "niederreiter_keygen.h"
+#include "H_Q_matrices_generation.h"
+#include "gf2x_arith_mod_xPplusOne.h"
+#include "rng.h"
+#include "dfr_test.h"
+#include <string.h>
+#include "randombytes.h"
+#include "nistseedexpander.h"
+/*----------------------------------------------------------------------------*/
+/* Implementation that should never be optimized out by the compiler */
```

Comment:
Why is this assumption necessary? Even volatile would not guarantee that this does not happen. In general, preventing something isn't optimised out can't be guaranteed in C.